lib/matplotlib/traitlets.py

```diff
@@ -0,0 +1,231 @@
+            return self._validate(conv)
+        trait.error(obj, value)
+
+class GSTransformInstance(TransformInstance,GetSetTraitType): pass
+
+
+
+class Callable(TraitType):
+    """A trait which is callable.
+
+    Notes
+    -----
+    Classes are callable, as are instances
+    with a __call__() method."""
+
+    info_text = 'a callable'
+
+    def validate(self, obj, value):
+        if callable(value):
+            return value
+        else:
+            self.error(obj, value)
+
+class ColorType(TraitType):
+    """A trait representing a color, can be either in RGB, or RGBA format.
+
+    Arguments:
+        force_rgb: bool: Force the return in RGB format instead of RGB. Default: False
+        as_hex: bool: Return the hex value instead. Default: False
+        default_alpha: float (0.0-1.0) or integer (0-255) default alpha value.
+
+    Accepts:
+        string: a valid hex  color string (i.e. #FFFFFF). 7 chars
+        tuple: a tuple of ints (0-255), or tuple of floats (0.0-1.0)
+        float: A gray shade (0-1)
+        integer: A gray shade (0-255)
+
+    Defaults: RGBA tuple, color black (0.0, 0.0, 0.0, 0.0)
+
+    Return:
+       A hex color string, a rgb or a rgba tuple. Defaults to rgba. When
+       returning hex string, the alpha property will be ignored. A warning
+       will be emitted if alpha information is passed different then 0.0
+
+    """
+    metadata = {
+        'force_rgb': False,
+        'as_hex' : False,
+        'default_alpha' : 0.0,
+        }
+    allow_none = True
+    info_text = 'float, int, tuple of float or int, or a hex string color'
+    default_value = (0.0,0.0,0.0,0.0)
+    named_colors = {}
+
+    def _int_to_float(self, value):
+        as_float = (np.array(value)/255).tolist()
+        return as_float
+
+    def _float_to_hex(self, value):
+        as_hex = '#%02x%02x%02x' % tuple([int(np.round(v * 255)) for v in\
+                                                                 value[:3]])
+        return as_hex
+
+    def _int_to_hex(self, value):
+        as_hex = '#%02x%02x%02x' % value[:3]
+        return as_hex
+
+    def _hex_to_float(self, value):
+        # Expects #FFFFFF format
+        split_hex = (value[1:3],value[3:5],value[5:7])
+        as_float = (np.array([int(v,16) for v in split_hex])/255.0).tolist()
+        return as_float
+
+    def _is_hex16(self, value):
+        try:
+            int(value, 16)
+            return True
+        except:
+            return False
+
+    def _float_to_shade(self, value):
+        grade = value*255.0
+        return (grade,grade,grade)
+
+    def _int_to_shade(self, value):
+        grade = value/255.0
+        return (grade,grade,grade)
+
+    def validate(self, obj, value):
+        in_range = False
+        if value is None or value is False or value in ['none','']:
+            # Return transparent if no other default alpha was set
+            return (0.0, 0.0, 0.0, 1.0)
+
+        if isinstance(value, float) and 0 <= value <= 1:
```

Comment:
There's a pull request to fix some things with the color traittype that i
probably forgot to include.
On Wed, Jul 22, 2015 at 11:03 PM Eric Firing <notifications@github.com>
wrote:

> In lib/matplotlib/traitlets.py
> <https://github.com/matplotlib/matplotlib/pull/4762#discussion_r35292090>:
>
> > +
> > +    def _float_to_shade(self, value):
> > +        grade = value*255.0
> > +        return (grade,grade,grade)
> > +
> > +    def _int_to_shade(self, value):
> > +        grade = value/255.0
> > +        return (grade,grade,grade)
> > +
> > +    def validate(self, obj, value):
> > +        in_range = False
> > +        if value is None or value is False or value in ['none','']:
> > +            # Return transparent if no other default alpha was set
> > +            return (0.0, 0.0, 0.0, 1.0)
> > +
> > +        if isinstance(value, float) and 0 <= value <= 1:
>
> This will eventually need to be modified, I expect:
>
> In [4]: isinstance(np.zeros((1,), dtype=np.float32)[0], float)False
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4762/files#r35292090>.
>